src/plugins/discover/public/application/main/components/sidebar/lib/visualize_trigger_utils.ts

```diff
@@ -53,6 +54,7 @@ export function triggerVisualizeActions(
     indexPatternId,
     fieldName: field.name,
     contextualFields,
+    originatingApp: PLUGIN_ID,
```

Comment:
Hopefully this is OK (😅). We use the same context with triggers from [Lens](https://github.com/elastic/kibana/blob/main/x-pack/plugins/lens/public/indexpattern_datasource/visualize_geo_field_button.tsx#L54) and [Data Visualizer](https://github.com/elastic/kibana/blob/main/x-pack/plugins/data_visualizer/public/application/common/components/field_data_row/action_menu/actions.ts#L94) plugins.

Reply:
We may need to dig more into this change.

The [PR that added UI counters to instrument when a geo field is visualized with visualizeGeoFieldAction](https://github.com/elastic/kibana/pull/123540) added `originatingApp`.

[Replace "Save and return" button with "Save" for Lens visualization created from Discover histogram or visualised field to avoid confusion](https://github.com/elastic/kibana/pull/129053) removed `originatingApp`.

Now we are adding `originatingApp`. Is adding `originatingApp` going to reintroduce https://github.com/elastic/kibana/issues/128695?

@DianaDerevyankina can you provide some context on how `originatingApp` was causing https://github.com/elastic/kibana/issues/128695?

Reply:
since @DianaDerevyankina is not longer working for EPAM/Elastic forwarding the question to @elastic/kibana-vis-editors

Reply:
Good catch @nreese , yes I think it would re-introduce this problem. The best idea I have to fix this is to ignore the `originatingApp` for "Visualize field" navigations (if `historyLocationState.type === ACTION_VISUALIZE_LENS_FIELD` holds true): https://github.com/elastic/kibana/blob/51816a06317040dcd6a41ae55cbbff606f75a414/x-pack/plugins/lens/public/app_plugin/mounter.tsx#L207

Feel free to add that logic in there - basically "if type is ACTION_VISUALIZE_LENS_FIELD, remove `originatingApp` from initial context"

Reply:
I can't think of a good use case where an app would want to receive the created Lens visualization when coming via "visualize field"